python/eos/analysis_file.py

```diff
@@ -157,42 +157,104 @@ def _sanitize_params(command, params):
         """Runs predefined analysis steps recorded in the analysis file."""
-        import inspect
-        command_map = {
-            'find-clusters':       eos.find_clusters,
-            'predict-observables': eos.predict_observables,
-            'sample-mcmc':         eos.sample_mcmc,
-            'sample-pmc':          eos.sample_pmc,
-        }
-
-        for idx, step in enumerate(self._steps):
-            if type(step) is not dict:
-                raise ValueError("Step #{} is not a key/value map.")
-
-            if 'command' not in step:
-                raise ValueError("Step #{} contains no command.")
+        from .tasks import _tasks
+        from itertools import chain, product
+        from inspect import signature
+        from collections import ChainMap
 
-            command  = step['command']
-            func     = command_map[command]
-            params   = step['parameters'] if 'parameters' in step else {}
-            params   = { params_map[(command, k)]: v for k, v in params.items() }
-            paramstr = ','.join(['{k}={v}'.format(k=k,v=v) for k, v in params])
+        def _expand_arguments(step_idx, task, arguments):
+            task_sig = signature(_tasks[task])
 
-            func_sig = inspect.signature(func)
-            func_required_args = {}
-            for n, p in func_sig.parameters.items():
-                if p.default() != p.empty():
+            task_required_args = {}
+            for n, p in task_sig.parameters.items():
+                if type(p.default) != p.empty:
                     continue
-                func_required_args += { n }
-            for n in func_required_args:
-                if n in params.keys():
+                task_required_args += { n }
+            for n in task_required_args:
+                if n in arguments.keys():
                     continue
-                eos.error('Mandatory argument \'{}\' not provided'.format(n))
-                return
+                raise ValueError(f'Task "{task}" in step #{step_idx} requires mandatory argument \'{n}\', which is not provided')
+
+            outer_list = []
+            for key, value in arguments.items():
+                if key not in task_sig.parameters:
+                    eos.warn(f'Task "{task}" does not expect argument {key}, possibly misspelled?')
+                    continue
+
+                param_type = task_sig.parameters[key].annotation
+                #print(f'{task}  {key}: {param_type}')
+
+                # straight match?
+                if type(value) is param_type:
+                    # append
+                    outer_list.append([{key: value}])
+                # handle integer parameters
+                elif param_type is int:
+                    # integer range?
+                    if type(value) is str and ('-' in value or ',' in value):
+                        values = list(chain.from_iterable(range(int(v.split("-")[0]),int(v.split("-")[-1])+1) for v in value.split(",")))
```

Comment:
Was this tested?

Reply:
Yes, but insufficiently. A full test suit should be added. Still thinking about how to do that. I think I would prefer returning the supposed ``eos-analysis`` commands as a string in a "dry-run" mode.

Reply:
I definitely like the idea of a "dry-run" mode

Reply:
Dry run and test added.